src/main/java/org/opensearch/flowframework/workflow/RegisterLocalModelStep.java

```diff
@@ -0,0 +1,198 @@
+                        )
+                    )
+                );
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Failed to register local model");
+                registerLocalModelFuture.completeExceptionally(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
+            }
+        };
+
+        String modelName = null;
+        String modelVersion = null;
+        String description = null;
+        MLModelFormat modelFormat = null;
+        String modelGroupId = null;
+        String modelContentHashValue = null;
+        String modelType = null;
+        String embeddingDimension = null;
+        FrameworkType frameworkType = null;
+        String allConfig = null;
+        String url = null;
+
+        for (WorkflowData workflowData : data) {
+            Map<String, Object> content = workflowData.getContent();
+
+            for (Entry<String, Object> entry : content.entrySet()) {
+                switch (entry.getKey()) {
+                    case NAME_FIELD:
+                        modelName = (String) content.get(NAME_FIELD);
+                        break;
+                    case VERSION_FIELD:
+                        modelVersion = (String) content.get(VERSION_FIELD);
+                        break;
+                    case DESCRIPTION_FIELD:
+                        description = (String) content.get(DESCRIPTION_FIELD);
+                        break;
+                    case MODEL_FORMAT:
+                        modelFormat = MLModelFormat.from((String) content.get(MODEL_FORMAT));
+                        break;
+                    case MODEL_GROUP_ID:
+                        modelGroupId = (String) content.get(MODEL_GROUP_ID);
+                        break;
+                    case MODEL_TYPE:
+                        modelType = (String) content.get(MODEL_TYPE);
+                        break;
+                    case EMBEDDING_DIMENSION:
+                        embeddingDimension = (String) content.get(EMBEDDING_DIMENSION);
+                        break;
+                    case FRAMEWORK_TYPE:
+                        frameworkType = FrameworkType.from((String) content.get(FRAMEWORK_TYPE));
+                        break;
+                    case ALL_CONFIG:
+                        allConfig = (String) content.get(ALL_CONFIG);
+                        break;
+                    case MODEL_CONTENT_HASH_VALUE:
+                        modelContentHashValue = (String) content.get(MODEL_CONTENT_HASH_VALUE);
+                        break;
+                    case URL:
+                        url = (String) content.get(URL);
+                        break;
+                    default:
+                        break;
+
+                }
+            }
+        }
+
+        if (Stream.of(
+            modelName,
+            modelVersion,
+            description,
```

Comment:
From [here](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/transport/register/MLRegisterModelInput.java#L94-L108), looks like only modelName, functionaName, modelFormat, modelConfig and url are mandatory fields not rest.

Reply:
model config is comprised of the `modelType` `embeddingDImension`, `frameworkType`, with an optional `all_config` field. We're not directly reading the model config from the `WorkflowData`, instead we check for the aforementioned fields and build the config in this step. 

This is based on the documentation ([link](https://opensearch.org/docs/latest/ml-commons-plugin/api/#request-fields)).

Additionally, there is no `function_name` for a local model registration, that is for the `RegisterRemoteModelStep`

Reply:
We should create ModelConfig object before the `if` condition then and handle the optional params of modelCondig before creating that object.